### ISSUE TITLE
[gatsby-remark-responsive-iframe] Handle case where iframe has 100% width

### ISSUE THREAD

**Issue by Undistraction** [gatsbyjs/gatsby#2468]
Although the plugin works nicely with Vimeo and YouTube, it breaks the Soundcloud embed completely, causing it to disappear. As you can see in the embed below, padding bottom is calculated as `NaN%`. 

```
<div class="gatsby-resp-iframe-wrapper" style="padding-bottom: NaN%; position: relative; height: 0; overflow: hidden;">
            <iframe scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/188212367&amp;color=%23ff5500&amp;auto_play=false&amp;hide_related=false&amp;show_comments=true&amp;show_user=true&amp;show_reposts=false&amp;show_teaser=true&amp;visual=true" style="
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
          "></iframe>
          </div>
```

 This is because all three variations of Soundcloud's embed use a width of `100%`.

```
<iframe width="100%" height="300" scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/188212367&amp;color=%23ff5500&amp;auto_play=false&amp;hide_related=false&amp;show_comments=true&amp;show_user=true&amp;show_reposts=false&amp;show_teaser=true&amp;visual=true"></iframe>
```

By default, the Soundcloud embed locks its height to one of three fixed heights (depending on the embed), then fills out its contain until it locks the width off at a maximum width. By hardcoding the width of the embed to a non-percentage value, the embed works, however the behaviour of the Soundcloud embed is impaired as it is now locked to a fixed aspect ratio. 

I think for situations like this, `gatsby-remark-responsive-iframe` needs to provide a way of opting an iframe out in its config - so in this situation I could tell it to work on all iframes other than Soundcloud. I think the easiest way would be to support a blacklist of domains which will be ignored if they are discovered in the iframe's `src` attribute. Something like:

```
{ 
  resolve: 'gatsby-remark-responsive-iframe',
  options: {
    ignore: ['https://w.soundcloud.com/player/']
  }
}
```

I'd be happy to take a run at this.

**Comment by KyleAMathews** [comment]
It sounds like it'd be pretty easy to just check for when width is "100%" and not try to rewrite the width/height there.

I'd love a PR for that.

Ignoring domains is a last-ditch way of solving things and I don't think it's necessary here.

**Comment by Undistraction** [comment]
I was thinking there will probably be other edge cases with other embeds and this would provide an escape hatch, but that would certainly solve the Soundcloud issue. I'll get something done in the next couple of days.

**Comment by Undistraction** [comment]
@KyleAMathews  Working on this now and have a couple of questions

1. Should we validate width and height to check they are unitless numbers and skip the transform if they aren't both valid?
2. If no to 1, how do you want to handle other percentage width values, for example `<iframe width="50%" height="400"></iframe>`, because at present any value other than a unitless number will break the embed. I would suggest we skip the transform.
3. If no to 1, how do you want to handle percentage height values, because at present any value other than a unitless number will break the embed.

I think 1 is the best solution - we should only allow unitless values for both width and height.

**Comment by KyleAMathews** [comment]
Yeah, 1 sounds great! If someone is setting percentages then the iframe is already "responsive" so we shouldn't be changing it.

**Comment by Undistraction** [comment]
Blocked at the moment by bug in Remark. If I skip the transform I get the original mAST back to compare against the snapshot. This is fine for an iframe, but is wrapped in `<p>` tags with text content of a newline for the object, making it hard to parse (and resulting in iffy markup). Will try and work around it.
